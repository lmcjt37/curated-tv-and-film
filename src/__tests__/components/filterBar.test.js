import React from 'react';
import { render, screen } from '@testing-library/react';

import FilterBar from '../../components/filterBar';

const mockHandleFilter = jest.fn();

it('render all initial child components for all', () => {
  var state = {
    years: [],
    filterResults: 'all',
    filterYear: 'All',
    filterGenre: {
      available: [
        'Action',
        'Adventure',
        'Animation',
        'Biography',
        'Comedy',
        'Crime',
        'Drama',
        'Family',
        'Fantasy',
        'History',
        'Horror',
        'Mystery',
        'Romance',
        'Sci-Fi',
        'Thriller'
      ],
      on: []
    },
    filterOrder: 'Ascending'
  };

  render(<FilterBar {...state} handleFilter={mockHandleFilter} />);

  expect(screen.getByTestId('filter-bar')).toBeTruthy();
  expect(screen.getAllByTestId('form-control-label').length).toEqual(3);
  expect(screen.queryByTestId('form-control-years')).toBeFalsy();
  expect(screen.getByTestId('form-control-sort')).toBeTruthy();
  expect(screen.getByTestId('chip-container')).toBeTruthy();
  expect(screen.getAllByTestId('chip').length).toEqual(15);
});

it('render all initial child components for movies', () => {
  var state = {
    years: [2017, 2018, 2019, 2020],
    filterResults: 'movies',
    filterYear: 'All',
    filterGenre: {
      available: [
        'Action',
        'Adventure',
        'Animation',
        'Biography',
        'Comedy',
        'Crime',
        'Drama',
        'Family',
        'Fantasy',
        'History',
        'Horror',
        'Mystery',
        'Romance',
        'Sci-Fi',
        'Thriller'
      ],
      on: []
    },
    filterOrder: 'Ascending'
  };

  render(<FilterBar {...state} handleFilter={mockHandleFilter} />);

  expect(screen.getByTestId('filter-bar')).toBeTruthy();
  expect(screen.getAllByTestId('form-control-label').length).toEqual(3);
  expect(screen.getByTestId('form-control-sort')).toBeTruthy();
  expect(screen.getByTestId('chip-container')).toBeTruthy();
  expect(screen.getAllByTestId('chip').length).toEqual(15);

  expect(screen.getByTestId('form-control-years')).toBeTruthy();
});

it('snapshot of initial component', () => {
  var state = {
    years: [2017, 2018, 2019, 2020],
    filterResults: 'movies',
    filterYear: 'All',
    filterGenre: {
      available: [
        'Action',
        'Adventure',
        'Animation',
        'Biography',
        'Comedy',
        'Crime',
        'Drama',
        'Family',
        'Fantasy',
        'History',
        'Horror',
        'Mystery',
        'Romance',
        'Sci-Fi',
        'Thriller'
      ],
      on: []
    },
    filterOrder: 'Ascending'
  };

  const { asFragment } = render(
    <FilterBar {...state} handleFilter={mockHandleFilter} />
  );
  expect(asFragment).toMatchSnapshot();
});
