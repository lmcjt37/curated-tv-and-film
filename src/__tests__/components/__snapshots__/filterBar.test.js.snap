// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot of initial component 1`] = `[Function]`;
